### PR TITLE
Install plugin command help

### LINKED ISSUE #2041: Add URL option to InstallPluginCommand help instructions
**Is your feature request related to a problem? Please describe.**
The `InstallPluginCommand` tool does not let user know they can use URL as an option to install the plugin from.

**Describe the solution you'd like**
1. Improve the output of `./bin/opensearch-plugin install --help` accordingly
2. Improve the online documentation as well (see below for details)

**Describe alternatives you've considered**
All considered alternatives seem to be less useful. 

**Additional context**

While the JavaDoc of [InstallPluginCommand](https://github.com/opensearch-project/OpenSearch/blob/1.2.4/distribution/tools/plugin-cli/src/main/java/org/opensearch/plugins/InstallPluginCommand.java) does [expliticly document use of URL option](https://github.com/opensearch-project/OpenSearch/blob/1.2.4/distribution/tools/plugin-cli/src/main/java/org/opensearch/plugins/InstallPluginCommand.java#L114) to install plugins from the [help instruction does not](https://github.com/opensearch-project/OpenSearch/blob/1.2.4/distribution/tools/plugin-cli/src/main/java/org/opensearch/plugins/InstallPluginCommand.java#L221).

For the reference this is the output of the command:

```sh
# Assuming OpenSearch 1.2.4
% ./bin/opensearch-plugin install --help
Install a plugin

The following official plugins may be installed by name:
  analysis-icu
  analysis-kuromoji
  analysis-nori
  analysis-phonetic
  analysis-smartcn
  analysis-stempel
  analysis-ukrainian
  discovery-azure-classic
  discovery-ec2
  discovery-gce
  ingest-attachment
  mapper-annotated-text
  mapper-murmur3
  mapper-size
  repository-azure
  repository-gcs
  repository-hdfs
  repository-s3
  store-smb
  transport-nio

Non-option arguments:
plugin id            

Option             Description                                            
------             -----------                                            
-E <KeyValuePair>  Configure a setting                                    
-b, --batch        Enable batch mode explicitly, automatic confirmation of
                     security permission                                  
-h, --help         Show help                                              
-s, --silent       Show minimal output                                    
-v, --verbose      Show verbose output
```

BTW: It is also missing in online Documentation: [Install OpenSearch / OpenSearch plugins](https://opensearch.org/docs/1.2/opensearch/install/plugins/) (I believe this contributes to UX issues like [this one](https://discuss.opendistrocommunity.dev/t/add-exporter-plugin-to-opensearch/7115/5)).

I suggest updating the **Non-option arguments** instructions to:

```
Non-option arguments:
plugin <name|URL>
```
Or something along these line based on <http://jopt-simple.github.io/jopt-simple/> conventions (ideally providing more info about how URL option can be used).

---

Notice: the command help instructions use an **"id"** in **Non-option arguments** part. Maybe a (nitpicking) detail but this can create an impression that besides `plugin name` there is also `plugin id`. The terminology is not consistent which can be confusing.

Actually, the internal class for
[PluginPropertiesExtension](https://opensearch.org/javadocs/1.2.4/OpenSearch/buildSrc/build/docs/javadoc/org/opensearch/gradle/plugin/PluginPropertiesExtension.html) uses the `name`. As far as I can tell all the other places in the code seem to refer to plugin `name` as opposed to plugin `id`. Even the _"sister"_ [RemovePluginCommand](https://github.com/opensearch-project/OpenSearch/blob/1.2.4/distribution/tools/plugin-cli/src/main/java/org/opensearch/plugins/RemovePluginCommand.java#L73) refers to `plugin name`.

On the other hand when the distribution is built the list of (built-in) plugins is [populated from plugins located under the plugins folder](https://github.com/opensearch-project/OpenSearch/blob/e3a44fa71b290fb265a94ef4297f044b9a63a762/server/build.gradle#L205). In this case the `plugin name` is based on folder name which is unique thus the terminology should be IMO leaning towards `plugin id` but that would result into code changes on many places.

### PR DESCRIPTION
**Pull request by joshpalis**
### Description
[Describe what this change achieves]
Improved InstallPluginCommand help instructions by adding URL to Non-argument options list. Also added unit tests to ensure  help instructions contain the correct output

### Issues Resolved
#2041 
 
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by VachaShah — COMMENTED** [review]
This looks good! Is it possible to add more details on how to use the plugin url in the help section?

**Comment by joshpalis** [comment]
@VachaShah Sure, I'll add more details about how to use plugin URLs

File: distribution/tools/plugin-cli/src/test/java/org/opensearch/plugins/InstallPluginCommandTests.java [hunk 1]
```diff
@@ -828,6 +828,31 @@ protected boolean addShutdownHook() {
         }
     }
 
+    public void testPluginsHelpNonOptionArgumentsOutput() throws Exception {
```
Comment (setiah) [inline distribution/tools/plugin-cli/src/test/java/org/opensearch/plugins/InstallPluginCommandTests.java:831]: Nice!

File: distribution/tools/plugin-cli/src/main/java/org/opensearch/plugins/InstallPluginCommand.java [hunk 2]
```diff
@@ -218,7 +218,7 @@
             Arrays.asList("b", "batch"),
             "Enable batch mode explicitly, automatic confirmation of security permission"
         );
-        this.arguments = parser.nonOptions("plugin id");
+        this.arguments = parser.nonOptions("plugin <name|URL>");
```
Comment (saratvemulapalli) [inline distribution/tools/plugin-cli/src/main/java/org/opensearch/plugins/InstallPluginCommand.java:221]: This is nice, we do also support installing by file. Could add that as an option as well?
For example: https://github.com/opensearch-project/opensearch-plugins/blob/main/INSTALLING.md#installing-opensearch-plugins

**Comment by setiah** [comment]
> 



> This looks good! Is it possible to add more details on how to use the plugin url in the help section?

Maybe stretching this a bit, but would it also make sense to add a warning message, for when a user tries to install a plugin from a non-standard (unverified) URL? Like what browser does when trying to install a dodgy (unverified) extension, just to intimate users. Just a thought!

File: distribution/tools/plugin-cli/src/main/java/org/opensearch/plugins/InstallPluginCommand.java (hunk 2 above)
Comment (joshpalis) [reply distribution/tools/plugin-cli/src/main/java/org/opensearch/plugins/InstallPluginCommand.java:221]: I've added more descriptions to the help command, as seen below. 
```
Install a plugin

Plugins are packaged as zip files. Each packaged plugin must contain a plugin properties file.

The install command takes a plugin id, which may be any of the following:
  An official opensearch plugin name
  Maven coordinates to a plugin zip
  A URL to a plugin zip

The following official plugins may be installed by name:
  analysis-icu
  analysis-kuromoji
  analysis-nori
  analysis-phonetic
  analysis-smartcn
  analysis-stempel
  analysis-ukrainian
  discovery-azure-classic
  discovery-ec2
  discovery-gce
  ingest-attachment
  mapper-annotated-text
  mapper-murmur3
  mapper-size
  repository-azure
  repository-gcs
  repository-hdfs
  repository-s3
  store-smb
  transport-nio

Non-option arguments:                 
[String] -- plugin <name|Zip File|URL>

Option             Description                                            
------             -----------                                            
-E <KeyValuePair>  Configure a setting                                    
-b, --batch        Enable batch mode explicitly, automatic confirmation of
                     security permission                                  
-h, --help         Show help                                              
-s, --silent       Show minimal output                                    
-v, --verbose      Show verbose output   
```

**Review by saratvemulapalli — APPROVED** [review]
Looks good to me.
A minor comment. Thanks @joshpalis !

**Comment by saratvemulapalli** [comment]
I'll wait for @VachaShah @setiah to chime in if they have any concerns merging this.

**Pull request merged**